go/runtime/registry/registry.go

```diff
@@ -410,8 +435,39 @@ func (r *runtimeRegistry) NewUnmanagedRuntime(ctx context.Context, runtimeID com
 	return newRuntime(ctx, runtimeID, r.cfg, r.consensus, r.logger)
 }
 
+func (r *runtimeRegistry) AddRoles(roles node.RolesMask, runtimeID *common.Namespace) error {
+	r.RLock()
+	defer r.RUnlock()
+
+	if runtimeID != nil {
+		rt, ok := r.runtimes[*runtimeID]
+		if !ok {
+			return fmt.Errorf("runtime/registry: runtime %s is not supported", *runtimeID)
+		}
+		rt.AddRoles(roles)
```

Comment:
Might be nicer to `reutrn nil` here early, and then the `else` part can be one level higher.